vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java

```diff
@@ -36,11 +46,15 @@
 
     @Autowired
     private IContentBlockManager contentBlockManager;
+    
+    @Autowired
+    private SpaceRepository spaceRepo;
 
     @RequestMapping("/staff/module/{moduleId}/slide/{id}")
-    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) {
+    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) throws JsonProcessingException {
 
         ISlide slide = slideManager.getSlide(id);
+        model.addAttribute("spaces", spaceRepo.findAll());
```

Comment:
this does not seem like a very good way to do that. what if there are 100 spaces? Better search spaces via an ajax call when a new slide is being created.